torch/nn/qat/modules/conv.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+from torch.nn.modules.conv import Conv2d as NNConv2d
+from torch.nn.modules.utils import _pair
+from torch.quantization.QConfig import default_qat_qconfig
+import torch.nn.functional as F
+
+class Conv2d(NNConv2d):
+    r"""
+    A Conv2d module attached with FakeQuantize modules for both output
+    activation and weight, used for quantization aware training.
+
+    We adopt the same interface as `torch.nn.Conv2d`, please see https://pytorch.org/docs/stable/nn.html?highlight=conv2d#torch.nn.Conv2d
+    for documentation.
+
+    Similar to `torch.nn.Conv2d`, with FakeQuantize modules initialized to
+    default.
+
+    Attributes:
+        observer: fake quant module for output activation, it's called observer
+            to align with post training flow, TODO: rename?
+        weight_fake_quant: fake quant module for weight
+
+    """
+
+    def __init__(self, in_channels, out_channels, kernel_size, stride=1,
+                 padding=0, dilation=1, groups=1,
+                 bias=True, padding_mode='zeros'):
+        super(Conv2d, self).__init__(in_channels, out_channels, kernel_size,
+                                     stride=stride, padding=padding, dilation=dilation,
+                                     groups=groups, bias=bias, padding_mode=padding_mode)
+        # fake quant module for output activation
+        self.observer = default_qat_qconfig.activation()
```

Comment:
nit - you can just make it an argument - then you don't need to do reassignment in from_float

Reply:
hmm, is there any value in keeping the API consistent with float conv module?

Reply:
Can we just call self.observer => self.fake_quant? I assume this is related to your poll, but having two different names (observer and fake_quant) and then using if /else or hasattr to determine what is present and then doing the appropriate action seems simpler to me.

Reply:
why having two names is simpler? I think worst case we can do self.observer_or_fake_quant and we don't need any of the hasattr checks.

Reply:
moved to constructor